src/Analyzers/CSharp/Analyzers/ConstantInterpolatedString/CSharpConstantInterpolatedStringAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConstantInterpolatedString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)] // constant interpolated strings is a C#-only feature.
+    internal class CSharpConstantInterpolatedStringAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpConstantInterpolatedStringAnalyzer() : base(
+            IDEDiagnosticIds.UseConstantInterpolatedStringDiagnosticId,
+            EnforceOnBuildValues.AddAccessibilityModifiers,
+            option: null,
+            new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_constant_interpolated_string), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!((CSharpCompilation)context.Compilation).LanguageVersion.HasConstantInterpolatedStrings())
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    var operation = (IBinaryOperation)context.Operation;
+                    var info = operation.SemanticModel!.GetSymbolInfo(operation.Syntax, context.CancellationToken);
```

Comment:
Is suppression safe here?

Reply:
Assuming this really is a nullable property:

```suggestion
                    var info = operation.GetRequiredSemanticModel().GetSymbolInfo(operation.Syntax, context.CancellationToken);
```

Reply:
@sharwell I can't find this extension method for IOperation. Seems to only exist for Document (otherwise, IntelliSense must be hanging locally).